ocs_ci/ocs/resources/pod.py

```
@@ -2582,25 +2582,76 @@ def check_toleration_on_pods(toleration_key=constants.TOLERATION_KEY):
         toleration_key (str): The toleration key to check
 
     """
-
     pod_objs = get_all_pods(
         namespace=config.ENV_DATA["cluster_namespace"],
         selector=[constants.TOOL_APP_LABEL],
         exclude_selector=True,
     )
-    flag = False
+
     for pod_obj in pod_objs:
         resource_name = pod_obj.name
-        tolerations = pod_obj.get().get("spec").get("tolerations")
-        for key in tolerations:
-            if key["key"] == toleration_key:
-                flag = True
-        if flag:
-            logger.info(f"The Toleration {toleration_key} exists on {resource_name}")
-        else:
+        if "-1-deploy" not in resource_name and "ocs-deviceset" not in resource_name:
+            pods_missing_toleration = []
+            tolerations = pod_obj.get().get("spec").get("tolerations")
+
+            # Check if any toleration matches the provided key
+            toleration_found = any(tol["key"] == toleration_key for tol in tolerations)
+
+            if not toleration_found:
+                logger.error(
+                    f"The pod {resource_name} does not have toleration {toleration_key}"
+                )
+                pods_missing_toleration.append(resource_name)
+            else:
+                logger.info(
+                    f"The Toleration {toleration_key} exists on {resource_name}"
+                )
+            if pods_missing_toleration:
+                logger.warning(
+                    f"Some pods are missing the toleration {toleration_key}: {', '.join(pods_missing_toleration)}"
+                )
+                raise AssertionError(
```

do you want to fail the testcase if toleration is not present ??